### DISCUSSION TITLE
Get smarts with explicit hydrogen counts

### DISCUSSION THREAD

**Discussion by phani-g** [rdkit/rdkit#5794] [Q&A]
Hi,

I am trying to get a smarts pattern with explicit hydrogens in it.

For example if I had benzene (c1ccccc1), I would like to get the smarts pattern [c;H1]1[c;H1][c;H1][c;H1][c;H1][c;H1]1 

How would I get it?

**Comment by vfscalfani** [comment]
Hi @phani-g

This is a little different than the pattern you specify there, but I saw this shared in a past RDKit UGM meeting (I forget the context of this query) and I wrote it down at the time:

```
>>> m = Chem.MolFromSmiles("c1ccccc1")
>>> Chem.MolToSmarts(Chem.AdjustQueryProperties(Chem.MergeQueryHs(Chem.AddHs(m))))
'[#6&!H0]1:[#6&!H0]:[#6&!H0]:[#6&!H0]:[#6&!H0]:[#6&!H0]:1'
```
Anyway, maybe this is useful for you to experiment with additional query parameters and find the right options that match what you are looking for. Here are all of the options:

https://www.rdkit.org/docs/source/rdkit.Chem.rdmolops.html?highlight=adjustquery#rdkit.Chem.rdmolops.AdjustQueryProperties

Vin
